docs/website/blog/2024-06-21-google-forms-to-notion.md

```diff
@@ -0,0 +1,142 @@
+
+## Implementation
+
+So here are a few steps followed:
+
+**Step 1: Link Google Form to Google Sheet**
+
+Link the Google Form to a Google Sheet to save responses in the sheet. Follow [Google's documentation](https://support.google.com/docs/answer/2917686?hl=en#zippy=%2Cchoose-where-to-store-responses) for setup.
+
+**Step 2: Google Apps Script for Data Transfer**
+
+Create a Google Apps Script to send data from Google Sheets to a Notion database via a webhook. This script triggers every time a form response is saved.
+
+**Google Apps Script code:**
+
+```text
+function sendWebhookOnEdit(e) {
+  var sheet = SpreadsheetApp.getActiveSpreadsheet().getActiveSheet();
+  var range = sheet.getActiveRange();
+  var updatedRow = range.getRow();
+  var lastColumn = sheet.getLastColumn();
+  var headers = sheet.getRange(1, 1, 1, lastColumn).getValues()[0];
+  var updatedFields = {};
+  var rowValues = sheet.getRange(updatedRow, 1, 1, lastColumn).getValues()[0];
+
+  for (var i = 0; i < headers.length; i++) {
+    updatedFields[headers[i]] = rowValues[i];
+  }
+
+  var jsonPayload = JSON.stringify(updatedFields);
+  Logger.log('JSON Payload: ' + jsonPayload);
+
+  var url = 'https://your-webhook.cloudfunctions.net/to_notion_from_google_forms'; // Replace with your Cloud Function URL
+  var options = {
+    'method': 'post',
+    'contentType': 'application/json',
+    'payload': jsonPayload
+  };
+
+  try {
+    var response = UrlFetchApp.fetch(url, options);
+    Logger.log('Response: ' + response.getContentText());
+  } catch (error) {
+    Logger.log('Failed to send webhook: ' + error.toString());
+  }
+}
+```
+
+**Step 3: Deploying the ETL Pipeline**
+
+Deploy a `dlt` pipeline to Google Cloud Functions to handle data transfer from Google Sheets to the Notion database. The pipeline is triggered by the Google Apps Script.
+
+1. Create a Google Cloud function.
+2. Create `main.py` with the Python code below.
+3. Ensure `requirements.txt` includes `dlt`.
+4. Deploy the pipeline to Google Cloud Functions.
+5. Use the function URL in the Google Apps Script.
+
+:::note
+This pipeline uses  `@dlt.destination` decorator which is used to set up custom destinations. Using custom destinations is a part of `dlt's` reverse ETL capabilities. To read more about `dlt's` reverse ETL pipelines, please read the [documentation here.](https://dlthub.com/docs/dlt-ecosystem/destinations/destination)
+:::
+
+**Python code for `main.py` (Google cloud functions) :**
+
+```py
+import dlt
+from dlt.common import json
+from dlt.common.typing import TDataItems
+from dlt.common.schema import TTableSchema
+from datetime import datetime
+from dlt.sources.helpers import requests
+
+@dlt.destination(name="notion", batch_size=1, naming_convention="direct", skip_dlt_columns_and_tables=True)
+def insert_into_notion(items: TDataItems, table: TTableSchema) -> None:
+    api_key = "your_notion_api_key"  # Replace with your Notion API Key
```

Comment:
do not suggest to put creds in code. Read from config, env, or credential vault.

Reply:
corrected